source/validate_id.cpp

```diff
@@ -2002,11 +2002,45 @@ bool idUsage::isValid<OpBranch>(const spv_instruction_t *inst,
                                 const spv_opcode_desc opcodeEntry) {}
 #endif
 
-#if 0
 template <>
-bool idUsage::isValid<OpBranchConditional>(
-    const spv_instruction_t *inst, const spv_opcode_desc opcodeEntry) {}
-#endif
+bool idUsage::isValid<SpvOpBranchConditional>(
+    const spv_instruction_t *inst, const spv_opcode_desc) {
+  const size_t numOperands = inst->words.size() - 1;
+  const size_t condOperandIndex = 1;
+  const size_t targetTrueIndex = 2;
+  const size_t targetFalseIndex = 3;
+
+  // num_operands is either 3 or 5 --- if 5, the last two need to be literal integers
+  if (numOperands != 3 &&
+      numOperands != 5) {
+    DIAG(0) << "OpBranchConditional requires either 3 or 5 parameters";
+    return false;
+  }
+
+  bool ret = true;
+
+  // grab the condition operand and check that it is a bool
+  auto condOp = module_.FindDef(inst->words[condOperandIndex]);
+  if (!condOp || !module_.IsBoolScalarType(condOp->type_id())) {
+    DIAG(0) << "Condition operand for OpBranchConditional must be of boolean type";
+    ret = false;
+  }
+
+  // target operands must be OpLabel
+  auto targetOpTrue = module_.FindDef(inst->words[targetTrueIndex]);
```

Comment:
Should also add a comment that validate_cfg.cpp checks that trueLabel and falseLabel are in the same function.  (It's the first check in PerformCfgChecks in that source file.)    So we don't need to check that condition here.